src/util/component-types.js

```diff
@@ -94,8 +94,12 @@ export function getFirst(props, types, defaultValue) {
 	return _.first(findTypes(props, types)) || defaultValue;
 }
 
-// omit props defined in propTypes of the given type
-// and any extra keys given in third argument
+// Omit props defined in propTypes of the given type and any extra keys given
+// in third argument
+//
+// We also include a "magic" prop that's always excluded called `callbackId`.
+// That prop can be used to identify a component in a list without having to
+// create extra closures.
 export function omitProps(props, type, keys = []) {
-	return _.omit(props, _.keys(type.propTypes).concat(keys).concat(['initialState']));
+	return _.omit(props, _.keys(type.propTypes).concat(keys).concat(['initialState', 'callbackId']));
```

Comment:
We talked about this a little bit, but if our goal is for this prop to cover all closure cases, I think it would be nice if the name were more general rather than tied to `id`.

Ideas: `callbackProp`, `callbackProps`, `callbackValue`.